distributed/event.py

```diff
@@ -0,0 +1,251 @@
+        with log_errors():
+            name = self._normalize_name(name)
+            # the default flag value is false
+            if name not in self._events:
+                return False
+
+            return self._events[name].is_set()
+
+    def _normalize_name(self, name):
+        """ Helper function to normalize an event name """
+        if isinstance(name, list):
+            name = tuple(name)
+
+        return name
+
+    def _get_or_create_event(self, name):
+        """ Helper function to return or create and return an event """
+        if name not in self._events:
+            # It seems we are the first one accessing it,
+            # so lets create a new event
+            self._events[name] = asyncio.Event()
+
+        return self._events[name]
+
+    def _delete_event(self, name):
+        """ Helper function to delete an event """
+        with suppress(KeyError):
+            del self._waiter_count[name]
+        with suppress(KeyError):
+            del self._events[name]
+
+
+class Event:
+    """ Distributed Centralized Event equivalent to asyncio.Event
+
+    An event stores a single flag, which is set to false on start.
+    The flag can be set to true (using the set() call) or back to false
+    (with the clear() call).
+    Every call to wait() blocks until the event flag is set to true.
+
+    Parameters
+    ----------
+    name: string (optional)
+        Name of the event.  Choosing the same name allows two
+        disconnected processes to coordinate an event.  If not given, a random
+        name will be generated.
+    client: Client (optional)
+        Client to use for communication with the scheduler.  If not given, the
+        default global client will be used.
+
+    Examples
+    --------
+    >>> event_1 = Event('a')  # doctest: +SKIP
+    >>> event_1.wait(timeout=1)  # doctest: +SKIP
+    >>> # in another process
+    >>> event_2 = Event('a')  # doctest: +SKIP
+    >>> event_2.set() # doctest: +SKIP
+    >>> # now event_1 will stop waiting
+    """
+
+    def __init__(self, name=None, client=None):
+        try:
+            self.client = client or Client.current()
+        except ValueError:
+            # Initialise new client
+            self.client = get_worker().client
+        self.name = name or "event-" + uuid.uuid4().hex
+
+    def wait(self, timeout=None):
+        """ Wait until the event is set.
+
+        Parameters
+        ----------
+        timeout : number, optional
+            Seconds to wait on the event in the scheduler.  This does not
+            include local coroutine time, network transfer time, etc..
+
+        Examples
+        --------
+        >>> event = Event('a')  # doctest: +SKIP
+        >>> event.wait(timeout=1)  # doctest: +SKIP
+
+        Returns
+        -------
+        True if the event was set of false, if a timeout happend
+        """
+        result = self.client.sync(
+            self.client.scheduler.event_wait, name=self.name, timeout=timeout,
```

Comment:
We might want to use `dask.utils.parse_timedelta` here.  I notice that this wasn't done in Lock (where I think you are copying from).  If you're interested in doing it there too that would be welcome.

(although all of this could also be done in a follow up).

This would allow for functionality like `await event.wait(timeout="100 ms")`

Reply:
Yes, that looks good.
I have created #3823 and will do this after this PR.